<?xml version="1.0"?>
<launch>
  <!-- Launch arguments -->
  <arg name="config_file" 
       default="$(find-pkg-share motor_control_app)/config/drive_component.yaml" 
       description="Path to the configuration file" />
  <arg name="serial_port" 
       default="/dev/ttyACM0" 
       description="Serial port for motor communication" />

  <!-- DriveComponent Node -->
  <node pkg="motor_control_app" 
        exec="drive_component_node" 
        name="drive_component" 
        output="screen"
        respawn="true" 
        respawn_delay="2.0">
    <param from="$(var config_file)" />
    <param name="serial_port" value="$(var serial_port)" />
  </node>
</launch>